src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const year4 = 'YYYY';
+  const year2 = 'YY';
+
+  const separatedDate = date.split(fromFormat[3]);
+
+  const dateData = {};
+
+  for (let i = 0; i < separatedDate.length; i++) {
+    dateData[fromFormat[i]] = separatedDate[i];
+  }
+
+  if (!dateData[year2]) {
+    dateData[year2] = dateData[year4].slice(-2);
+  } else {
+    if (dateData[year2] < 30) {
+      dateData[year4] = `20${dateData[year2]}`;
+    } else {
+      dateData[year4] = `19${dateData[year2]}`;
+    }
+  }
```

Comment:
The logic for converting the year part of the date seems to be reversed. The conversion from 'YY' to 'YYYY' should only occur if the 'YY' format is present in the `fromFormat` array, and vice versa. You need to determine which year format is in the `fromFormat` and which is in the `toFormat`, and then perform the conversion accordingly.